db/external_sst_file_test.cc

```diff
@@ -4146,6 +4080,458 @@ TEST_P(IngestDBGeneratedFileTest2, NotOverlapWithDB) {
+        ASSERT_GT(file_meta->smallest_seqno, 0);
+        ASSERT_GE(file_meta->largest_seqno, file_meta->smallest_seqno);
+        sst_file_paths.emplace_back(file_meta->directory + "/" +
+                                    file_meta->relative_filename);
+      }
+    }
+    from_db->ReleaseSnapshot(snapshot);
+
+    Status s;
+    // Perform ingestion and validate results
+    if (multi_level_ingestion && options.num_levels > 1) {
+      // fail_if_bottommost requres ingesting all files into the last level,
+      // so it fails if we are assiging files to multiple levels.
+      ingest_opts.fail_if_not_bottommost_level = true;
+      s = db_->IngestExternalFile(non_overlap_cf, sst_file_paths, ingest_opts);
+      ASSERT_NOK(s);
+      ASSERT_TRUE(s.ToString().find("Files cannot be ingested to Lmax") !=
+                  std::string::npos);
+      ingest_opts.fail_if_not_bottommost_level = false;
+    }
+    if (ingest_opts.snapshot_consistency) {
+      // snapshot_consisteny requires global sequence number assignment to
+      // ingested files if there is any live snapshot.
+      snapshot = db_->GetSnapshot();
+      s = db_->IngestExternalFile(non_overlap_cf, sst_file_paths, ingest_opts);
+      ASSERT_NOK(s);
+      ASSERT_TRUE(s.ToString().find(
+          "An ingested file is assigned to a non-zero sequence number"));
+      db_->ReleaseSnapshot(snapshot);
+    }
+    ASSERT_OK(
+        db_->IngestExternalFile(non_overlap_cf, sst_file_paths, ingest_opts));
+
+    // Validate ingested data.
+    ReadOptions ro;
+    std::string val;
+    for (int k = 0; k < 100; ++k) {
+      s = db_->Get(ro, handles_[1], Key(k), &val);
+      ASSERT_OK(s) << "Should find ingested key " << Key(k);
+      ASSERT_EQ(val, expected_values[k]) << "key: " << Key(k);
+    }
+
+    // Some ingestion failure cases
+    IngestExternalFileOptions restrictive_opts = ingest_opts;
+    restrictive_opts.allow_db_generated_files = false;
+    s = db_->IngestExternalFile(non_overlap_cf, {sst_file_paths[0]},
+                                restrictive_opts);
+    ASSERT_NOK(s);
+    ASSERT_TRUE(s.ToString().find("External file version not found") !=
+                std::string::npos);
+
+    // Ingesting the same file should fail due to overlap with CF
+    s = db_->IngestExternalFile(non_overlap_cf, {sst_file_paths[0]},
+                                ingest_opts);
+    ASSERT_NOK(s);
+    ASSERT_TRUE(
+        s.ToString().find(
+            "An ingested file is assigned to a non-zero sequence number") !=
+        std::string::npos)
+        << s.ToString();
+
+    // Overlap with data in the CF
+    if (ingest_opts.allow_blocking_flush) {
+      s = db_->IngestExternalFile(overlap_cf, sst_file_paths, ingest_opts);
+
+      ASSERT_NOK(s);
+      ASSERT_TRUE(
+          s.ToString().find(
+              "An ingested file is assigned to a non-zero sequence number") !=
+          std::string::npos)
+          << s.ToString();
+    }
```

Comment:
Why is the difference between this portion of the test with below in TEST_P(IngestDBGeneratedFileTest, FailureCase) ?
```
const std::string err =
        "An ingested file is assigned to a non-zero sequence number, which is "
        "incompatible with ingestion option allow_db_generated_files";
    ingest_opts.allow_db_generated_files = true;
    s = db_->IngestExternalFile(to_ingest_files, ingest_opts);
    ASSERT_TRUE(s.ToString().find(err) != std::string::npos);
    ASSERT_NOK(s);
```